test/jdk/javax/swing/JPopupMenu/SetInvokerJPopupMenuTest.java

```diff
@@ -0,0 +1,159 @@
+                SwingUtilities.invokeAndWait(() -> {
+                    jtb = new MyButton("Press Me");
+                    jtb.addActionListener(new ActionListener( ) {
+                        @Override
+                        public void actionPerformed(ActionEvent ev) {
+                            if (!popup.isVisible()) {
+                                postUp();
+                            }
+                            else {
+                                postDown();
+                            }
+                        }
+                    });
+
+                    f = new JFrame( );
+                    f.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+                    f.setSize(300, 400);
+                    Container c = f.getContentPane( );
+                    c.setLayout(new FlowLayout( ));
+                    c.add(jtb);
+                    f.setVisible(true);
+                    popup = new MyPopupMenu();
+                    popup.add(new JMenuItem("A"));
+                    popup.add(new JMenuItem("B"));
+                    popup.add(new JMenuItem("C"));
+                    popup.setVisible(false);
+                    p = jtb.getLocationOnScreen();
+                });
+
+                robot.waitForIdle();
+                robot.setAutoDelay(50);
+                robot.mouseMove(p.x + 15, p.y + 15);
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+                robot.waitForIdle();
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+            } catch (UnsupportedLookAndFeelException e) {
+                System.out.println("Note: LookAndFeel " + LF.getClassName()
+                        + " is not supported on this configuration");
+            } finally {
+                if (f != null) {
+                    SwingUtilities.invokeAndWait(() -> f.dispose());
+                }
+            }
+            if (popup.isVisible()) {
+                throw new RuntimeException("PopupMenu is not taken down after"+
+                        "single button click");
+            }
+        }
+    }
+
+    public static void postUp() {
+        popup.setInvoker(jtb);
+        popup.setVisible(true);
+    }
+
+    public static void postDown() {
+        popup.setVisible(false);
+    }
+
+    private static class MyButton extends JButton {
+        public MyButton(String string) {
+            super (string);
+        }
+        @Override
+        protected void processMouseEvent(MouseEvent e) {
+            super.processMouseEvent(e);
+        }
+    }
+
+    private static class MyPopupMenu extends JPopupMenu {
+        private static final long serialVersionUID = 1L;
+        @Override
+        public void setVisible( boolean state ) {
+            if( !state ) {
+                Exception ex = new Exception();
```

Comment:
It is unclear what is the purpose of this method?

Reply:
This method is needed to reproduce the issue.
If we remove this method then the popup menu never goes off the screen after the the first mouse click.